src/core/packages/pricing/common/src/pricing_tiers_client.ts

```diff
@@ -87,4 +87,40 @@ export class PricingTiersClient implements IPricingTiersClient {
 
     return false;
   };
+
+  public product = (): PricingProduct | undefined => {
+    if (this.tiers.enabled === false || this.tiers.products == null) {
+      return undefined;
+    }
+
+    if (this.tiers.products[0].name === 'observability') {
+      return {
+        type: 'observability' as const,
+        tier: this.tiers.products[0].tier as 'complete' | 'logs_essentials',
+      };
+    } else {
+      // Assert all products are security-related product lines
+      const securityProductLines = ['ai_soc', 'endpoint', 'cloud', 'security'] as const;
+      const allProductsAreSecurityRelated = this.tiers.products.every((product) =>
+        securityProductLines.includes(product.name as any)
+      );
+
+      if (!allProductsAreSecurityRelated) {
+        throw new Error(
+          'Mixed product types detected: all products must be either observability or security-related'
+        );
+      }
+
+      // Note: We can safely use products[0].tier here because the schema validates
+      // that all products have the same tier
+      return {
+        type: 'security' as const,
+        tier: this.tiers.products[0].tier as 'search_ai_lake' | 'complete' | 'essentials',
```

Comment:
We could use `PricingProductSecurity['tier']`

Reply:
actually I didn't need to cast anyway, so just removed the `as ...`